lib/facility_factory.rb

```diff
@@ -0,0 +1,55 @@
+class FacilityFactory
+
+    def create_facilities(locations)
+        if locations[0][:state] == "CO"
+            create_co_facilities(locations)
+        elsif locations[0][:state] == "NY"
+            create_ny_facilities(locations)
+        elsif locations[0][:state] == "MO"
+            create_mo_facilities(locations)
+        end
+    end
+
+    def create_co_facilities(locations)
+        facilities_array = []
+        locations.each do |location|
+            branch_details = {
+                name:location[:dmv_office],
+                address: "#{location[:address_li]} #{location[:address__1]}, #{location[:city]}, #{location[:state]} #{location[:zip]}",
+                phone:location[:phone]
+            }
+            facility = Facility.new(branch_details)
+            facilities_array << facility 
+        end
+        facilities_array
```

Comment:
Yup, this looks great! You could also have an instance variable holding the facilities_array stored in an initialize, that way that data is always stored and can be accessed repeatedly, instead of having to run this method everytime you want the data.